File: python/Petshop.py
class PetShop:
    products = []

    def __init__(self, id, name, category, price):
        self.id = id
        self.name = name
        self.category = category
        self.price = price
        PetShop.products.append(self)

    @staticmethod
    def add_product(id, name, category, price):
        PetShop(id, name, category, price)
        print("Produk berhasil ditambahkan!")

    @staticmethod
    def display_products():
        if not PetShop.products:
            print("Tidak ada produk dalam daftar.")
            return
        print("\nDaftar Produk:")
        for product in PetShop.products:
            print(f"ID: {product.id}, Nama: {product.name}, Kategori: {product.category}, Harga: {product.price}")

    @staticmethod
    def search_product(name):
        found = False
        for product in PetShop.products:
            if product.name.lower() == name.lower():
                print(f"Ditemukan: ID: {product.id}, Nama: {product.name}, Kategori: {product.category}, Harga: {product.price}")
                found = True
        if not found:
            print(f"Produk dengan nama {name} tidak ditemukan.")

    @staticmethod
    def update_product(id, new_name, new_category, new_price):
        for product in PetShop.products:
            if product.id == id:
                product.name = new_name
                product.category = new_category
                product.price = new_price
                print("Produk berhasil diperbarui.")
                return
        print(f"Produk dengan ID {id} tidak ditemukan.")

    @staticmethod
    def delete_product(id):
        for product in PetShop.products:
            if product.id == id:
                PetShop.products.remove(product)
                print("Produk berhasil dihapus.")
                return
        print(f"Produk dengan ID {id} tidak ditemukan.")


def main():
    while True:
        print("\nMenu:")
        print("1. Tambah Produk")
        print("2. Tampilkan Produk")
        print("3. Cari Produk")
        print("4. Ubah Produk")
        print("5. Hapus Produk")
        print("6. Keluar")
        choice = input("Pilihan: ")

        if choice == '1':
            id = int(input("Masukkan ID Produk: "))
            name = input("Masukkan Nama Produk: ")
            category = input("Masukkan Kategori: ")
            price = float(input("Masukkan Harga: "))
            PetShop.add_product(id, name, category, price)
        elif choice == '2':
            PetShop.display_products()
        elif choice == '3':
            name = input("Masukkan Nama produk yang dicari: ")
            PetShop.search_product(name)
        elif choice == '4':
            id = int(input("Masukkan ID produk yang ingin diubah: "))
            new_name = input("Masukkan Nama Baru: ")
            new_category = input("Masukkan Kategori Baru: ")
            new_price = float(input("Masukkan Harga Baru: "))
            PetShop.update_product(id, new_name, new_category, new_price)
        elif choice == '5':
            id = int(input("Masukkan ID produk yang ingin dihapus: "))
            PetShop.delete_product(id)
        elif choice == '6':
            print("Keluar dari program.")
            break
        else:
            print("Pilihan tidak valid. Silakan coba lagi.")


if __name__ == "__main__":
    main()

File: java/Petshop.java
import java.util.ArrayList;
import java.util.Scanner;

class PetShop {
    private int id;
    private String name;
    private String category;
    private double price;
    private static ArrayList<PetShop> products = new ArrayList<>();

    public PetShop(int id, String name, String category, double price) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.price = price;
        products.add(this);
    }

    public int getId() { return id; }
    public String getName() { return name; }
    public String getCategory() { return category; }
    public double getPrice() { return price; }

    public void setName(String newName) { this.name = newName; }
    public void setCategory(String newCategory) { this.category = newCategory; }
    public void setPrice(double newPrice) { this.price = newPrice; }

    public static void addProduct(int id, String name, String category, double price) {
        new PetShop(id, name, category, price);
        System.out.println("Produk berhasil ditambahkan!");
    }

    public static void displayProducts() {
        if (products.isEmpty()) {
            System.out.println("Tidak ada produk dalam daftar.");
            return;
        }
        System.out.println("\nDaftar Produk:");
        for (PetShop product : products) {
            System.out.println("ID: " + product.getId() + ", Nama: " + product.getName() +
                               ", Kategori: " + product.getCategory() + ", Harga: " + product.getPrice());
        }
    }

    public static void searchProduct(String name) {
        boolean found = false;
        for (PetShop product : products) {
            if (product.getName().equalsIgnoreCase(name)) {
                System.out.println("Ditemukan: ID: " + product.getId() + ", Nama: " + product.getName() +
                                   ", Kategori: " + product.getCategory() + ", Harga: " + product.getPrice());
                found = true;
            }
        }
        if (!found) {
            System.out.println("Produk dengan nama " + name + " tidak ditemukan.");
        }
    }

    public static void updateProduct(int id, String newName, String newCategory, double newPrice) {
        for (PetShop product : products) {
            if (product.getId() == id) {
                product.setName(newName);
                product.setCategory(newCategory);
                product.setPrice(newPrice);
                System.out.println("Produk berhasil diperbarui.");
                return;
            }
        }
        System.out.println("Produk dengan ID " + id + " tidak ditemukan.");
    }

    public static void deleteProduct(int id) {
        for (int i = 0; i < products.size(); i++) {
            if (products.get(i).getId() == id) {
                products.remove(i);
                System.out.println("Produk berhasil dihapus.");
                return;
            }
        }
        System.out.println("Produk dengan ID " + id + " tidak ditemukan.");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("\nMenu:");
            System.out.println("1. Tambah Produk");
            System.out.println("2. Tampilkan Produk");
            System.out.println("3. Cari Produk");
            System.out.println("4. Ubah Produk");
            System.out.println("5. Hapus Produk");
            System.out.println("6. Keluar");
            System.out.print("Pilihan: ");

            int choice = scanner.nextInt();
            scanner.nextLine();  // Buang newline

            switch (choice) {
                case 1:
                    System.out.print("Masukkan ID Produk: ");
                    int id = scanner.nextInt();
                    scanner.nextLine();
                    System.out.print("Masukkan Nama Produk: ");
                    String name = scanner.nextLine();
                    System.out.print("Masukkan Kategori: ");
                    String category = scanner.nextLine();
                    System.out.print("Masukkan Harga: ");
                    double price = scanner.nextDouble();
                    addProduct(id, name, category, price);
                    break;
                case 2:
                    displayProducts();
                    break;
                case 3:
                    System.out.print("Masukkan Nama produk yang dicari: ");
                    String searchName = scanner.nextLine();
                    searchProduct(searchName);
                    break;
                case 4:
                    System.out.print("Masukkan ID produk yang ingin diubah: ");
                    int updateId = scanner.nextInt();
                    scanner.nextLine();
                    System.out.print("Masukkan Nama Baru: ");
                    String newName = scanner.nextLine();
                    System.out.print("Masukkan Kategori Baru: ");
                    String newCategory = scanner.nextLine();
                    System.out.print("Masukkan Harga Baru: ");
                    double newPrice = scanner.nextDouble();
                    updateProduct(updateId, newName, newCategory, newPrice);
                    break;
                case 5:
                    System.out.print("Masukkan ID produk yang ingin dihapus: ");
                    int deleteId = scanner.nextInt();
                    deleteProduct(deleteId);
                    break;
                case 6:
                    System.out.println("Keluar dari program.");
                    scanner.close();
                    return;
                default:
                    System.out.println("Pilihan tidak valid. Silakan coba lagi.");
            }
        }
    }
}
